hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/QueryUtil.java

```diff
@@ -267,15 +264,7 @@ private void initContext(InternalSerializationService iss) {
                 return;
             }
 
-            SecurityContext securityContext = node.securityContext;
-            if (securityContext != null) {
-                assert subject != null : "Missing subject when security context exists";
-                this.ssc = securityContext.createSqlContext(subject);
-            } else {
-                this.ssc = NoOpSqlSecurityContext.INSTANCE;
-            }
-
-            this.evalContext = ExpressionEvalContext.createContext(arguments, node.getNodeEngine(), iss, ssc);
+            this.evalContext = new UntrustedExpressionEvalContext(arguments, iss, node.nodeEngine);
```

Comment:
I think would should also remove `subject` completely from JoinProjection. It is not used and I just realized that serialization of it would be a breaking change (causing query failures during rolling upgrade).

Reply:
hmm, I can add comment, but not resolve a discussion